### DISCUSSION TITLE
support Candence Tensilica HIFI4 DSP on libmetal

### DISCUSSION THREAD

**Discussion by mrzhuyuanheng** [OpenAMP/libmetal#174] [Q&A] [answered]
hi!
Is there any plan to support Candence Tensilica HIFI4 DSP processor on libmetal?
if i want to suport this processor myself, how should i do ?

**Comment by edmooring** [comment]
There is a set of porting guidelines at https://github.com/OpenAMP/open-amp/wiki/OpenAMP-Porting-GuideLine 
These are a good starting point. What is your use case? I don't see any support for the Cadence HiFi4 on the Linux kernel remoteproc side.

**Comment by mrzhuyuanheng** [comment]
thanks !

I'm new to open-amp. I want to support for the Cadence HiFi4 on the RTOS remoteproc side rather than linux.
my environment lists as follow:
- processor: similiar to [NXP i.MXRT600](https://www.nxp.com/products/processors-and-microcontrollers/arm-microcontrollers/i-mx-rt-crossover-mcus/i-mx-rt600-crossover-mcu-with-arm-cortex-m33-and-dsp-cores:i.MX-RT600) which also have a AP core (Arm® Cortex®-M33) and CP core (Cadence® Tensilica® HiFi 4 audio DSP core)
- system: AP(use zephyr),  CP(use zephyr or xos(like freeRTOS))

I want to use open-amp on both AP and CP.  AP and CP all use RTOS system. and i'm evaluating the feasibility.

since libmetal support arm processor, AP can directly use libmetal and open-amp on zephyr. AP processor is Tensilica, which libmetal don't support, so i want to porting libmetal on it.

how do you think it? Do you have any good suggestions?

**Comment by xiaoxiang781216** [comment]
We have run OpenAMP on HiFi4 DSP and connect to Cortex M process in a RTOS environment. If you are interesting, we can upstream the change(a small patch:)).

**Reply by mrzhuyuanheng** [reply]
@xiaoxiang781216 thanks for you answer ! That would be great !
I'm interesting in it. please upstream the patch

**Comment by xiaoxiang781216** [comment, marked as answer]
See here: https://github.com/OpenAMP/libmetal/pull/175
